Codecademy/gamut
# feat(Tooltip): Add tooltip placements

### Overview

This is all work related to refactoring ToolTips + adding a placement prop to allow them to break out of their context


* Adds 'sml' Tooltip-esque size variant to Popover
* Add `animation` and `skipFocusTrap` to popovers to allow more flexibility (partially to better mimic ToolTips)
* Add 'background-contrast' to our semantic colors (black for DarkMode, white for Light)
* Adds 'placement' prop to allow Tooltips to break out of their context if needed
* Refactors ToolTips + Popovers to use newer variant syntax + clean up file structure
* Adds related tests
* I also ending up adding a VS Code extension setting here for[ Story Explorer ](https://marketplace.visualstudio.com/items?itemName=joshbolduc.story-explorer) but can move that to a different PR 😅 


### PR Checklist

- [x] Related to JIRA ticket: WEB-2005

#### Testing Instructions
* Check that Tooltips and Popover still work as expected (both with VO and regularly) on their respective storybook pages
* Check that new 'floating' Tooltip placement breaks out of its context in the example in Storybook

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/31380)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/2178)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Author | [Another Link](https://github.com/codecademy-engineering/author/pull/2156) | [Author](http://www.google.fr/ 'Named link title')  |
